### PR TITLE
[vcpkg] vcpkg_configure_meson: cross-compile fixes for Android

### PR DESCRIPTION
**Pull request by jwtowner**
This PR includes two fixes to the meson cross-compile config.

The first change fixes the regex used to partition linker flag options so that it supports double-hyphen prefixes. Without this change, passing a linker flag such as "--target=armv7a-linux-androideabi21" will end up being passed incorrectly as "- -target=armv7a-linux-androideabi21".

The second change ensures that `vcpkg_internal_meson_generate_cross_file` always generates a cross file when the target platform is either Android or iOS, even if the host and target CPU architectures are the same. For example, when cross compiling harfbuzz:x64-android on a Windows 64-bit (x86-64) host, `vcpkg_internal_meson_generate_cross_file` normally does not generate a cross file because the host and target CPU architectures are equal, and this in turn causes Meson to attempt to run the generated x64-android sanity checking binaries as if they were Windows binaries, which causes the configuration to fail altogether.

One question, is there a better way to check if `VCPKG_CMAKE_SYSTEM_NAME` does not match the host system, and in a way that will be future proofed for other cross-compile target systems other than Android and iOS?

This PR, in tandem with PR #15605, allows for harfbuzz (and potentially other ports which also use meson) to be built for arm-android, arm64-android, x86-android and x64-android. Despite harfbuzz:*-android also relying on #15605, this PR should be safe to merge independently of it.

### CODE REVIEW

File: scripts/cmake/vcpkg_configure_meson.cmake [hunk 1]
```diff
@@ -128,7 +128,7 @@ function(vcpkg_internal_meson_generate_native_file_config _config) #https://meso
     else()
         set(LINKER_FLAGS_${_config} "${VCPKG_DETECTED_CMAKE_STATIC_LINKER_FLAGS_${_config}}")
     endif()
-    string(REGEX REPLACE "( |^)(-|/)" ";\\2" LINKER_FLAGS_${_config} "${LINKER_FLAGS_${_config}}")
+    string(REGEX REPLACE "( |^)(--?|/)" ";\\2" LINKER_FLAGS_${_config} "${LINKER_FLAGS_${_config}}")
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_configure_meson.cmake:131]: Use `vcpkg_internal_meson_convert_compiler_flags_to_list` instead

File: scripts/cmake/vcpkg_configure_meson.cmake [hunk 2]
```diff
@@ -316,7 +316,7 @@ function(vcpkg_internal_meson_generate_cross_file_config _config) #https://meson
...
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_configure_meson.cmake:319]: same as above

File: scripts/cmake/vcpkg_configure_meson.cmake (hunk 1 above)
Comment (jwtowner) [reply scripts/cmake/vcpkg_configure_meson.cmake]: Fixed in 8242f9e

File: scripts/cmake/vcpkg_configure_meson.cmake (hunk 2 above)
Comment (jwtowner) [reply scripts/cmake/vcpkg_configure_meson.cmake]: Fixed in 8242f9e

**Review by PhoebeHui — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Awesome, thanks for the fixes!

**Pull request merged**